ts/features/wallet/cobadge/component/CoBadgeCard.tsx

```diff
@@ -3,27 +3,60 @@ import { getCardIconFromBrandLogo } from "../../../../components/wallet/card/Log
 import { CreditCardPaymentMethod } from "../../../../types/pagopa";
 import { dateFromMonthAndYear } from "../../../../utils/dates";
 import { isPaymentMethodExpired } from "../../../../utils/paymentMethod";
+import I18n from "../../../../i18n";
+import { localeDateFormat } from "../../../../utils/locale";
 import BaseCoBadgeCard from "./BaseCoBadgeCard";
 
 type Props = { enhancedCoBadge: CreditCardPaymentMethod };
 
+/**
+ * Generate the accessibility label for the card.
+ */
+const getAccessibilityRepresentation = (
+  enhancedCoBadge: CreditCardPaymentMethod,
+  expiringDate?: Date
+) => {
+  const cardRepresentation = I18n.t("wallet.accessibility.folded.coBadge", {
+    brand: enhancedCoBadge.info.brand,
+    bankName:
+      enhancedCoBadge.abiInfo?.name ??
+      I18n.t("wallet.accessibility.folded.bankNotAvailable")
+  });
+
+  const computedValidity =
+    typeof expiringDate !== "undefined"
+      ? `, ${I18n.t("cardComponent.validUntil")} ${localeDateFormat(
+          expiringDate,
+          I18n.t("global.dateFormats.numericMonthYear")
+        )}`
+      : "";
+
+  return `${cardRepresentation}${computedValidity}`;
+};
+
 /**
  * Render a Co-badge card already added to the wallet
  * @param props
  * @constructor
  */
 const CoBadgeCard: React.FunctionComponent<Props> = props => {
   const brandLogo = getCardIconFromBrandLogo(props.enhancedCoBadge.info);
+  const expiringDate = dateFromMonthAndYear(
+    props.enhancedCoBadge.info.expireMonth,
+    props.enhancedCoBadge.info.expireYear
+  ).toUndefined();
+
   return (
     <BaseCoBadgeCard
       abi={props.enhancedCoBadge.abiInfo ?? {}}
       isExpired={isPaymentMethodExpired(props.enhancedCoBadge).getOrElse(false)}
-      expiringDate={dateFromMonthAndYear(
-        props.enhancedCoBadge.info.expireMonth,
-        props.enhancedCoBadge.info.expireYear
-      ).toUndefined()}
+      expiringDate={expiringDate}
       brandLogo={brandLogo}
       caption={props.enhancedCoBadge.caption}
+      accessibilityLabel={getAccessibilityRepresentation(
+        props.enhancedCoBadge,
+        expiringDate
+      )}
```

Comment:
Like the previous comment for the `BaseBancomatCard`, also in this case `BaseCobadgeCard` is used by `PreviewCoBadgeCard`. Moving the accessibility logic in `BaseCobadgeCard` could make accessible also the `PreviewCoBadgeCard`. What you think?

Reply:
You're right! About this and the previous comment I thought I didn't have all the informations (I just added the `brand` prop for the CoBadge card). It should be fixed in the latest commit 🙂

N.B. Props like `brand` and `bankName` could be `undefined` following the data structure for the CoBadge card. I just added `""` as a fallback, let me know if we usually handle this in any other way inside the app!

Reply:
Don't worry, I totally understand, the payment methods have some variant due to the different shape of data that we have to handle and they are not documented anywhere.
Btw great job! With the latest change all the payment methods accessibility is fixed 🚀